tutorials/image/cifar10_estimator/cifar10_main.py

```diff
@@ -159,10 +160,9 @@ def _resnet_model_fn(features, labels, mode, params):
           learning_rate=learning_rate, momentum=momentum)
 
       chief_hooks = []
-      if sync:
+      if params.sync:
```

Comment:
I made `sync` an argument of the enclosing function because it shouldn't be hyperparameter tuned right? I'm not sure if maybe someone would want to let an hptuner decide on whether or not to use synchronous gradient application, but from a code perspective I wanted `params` to only contain tunable parameters, and everything else to be hardcoded in the enclosing function scope.

Reply:
@martinwicke mentioned we want to put everything into params.  Params is not not just for hyper tuning based on my understanding of the field from Martin.  I could be very wrong.  This might be a great place to get agreement before I remove the rest of the method args.

Reply:
Yeah, there's no strong preference for params to only be hyperparameters. In this case, putting these arguments into the params object allows us to save a level of indirection and make the code more readable. Some of the parameters are legitimate hyperparameters, some are not, but the code is going to be better if we remove the extra create_ function.

Reply:
I'm pretty ambivalent between the two approaches. But IIUC under the implementation being suggested, if you were to try to supply the `tf.HParams` given to learn_runner with a `tf.Tuner`, the Tuner would have to also supply things like `sync` because the hparams are passed through directly to the constructed `Estimator` within the `experiment_fn`, so causing this to be `params.sync` means any `tf.Tuner` must be aware of your choice of synchronous gradient application, which seems not great.

Reply:
You either have to 
- Make sure that things like sync have sensible defaults (better)
- Update the hparams supplied with things you know you want before passing it on (ok)
- Make the tuner supply things like sync (worst)

I think making sure that things have good defaults if a param isn't supplied is ideal. Only for hyper-parameters for which defaults are not possible would it be ok to error out.

Reply:
Not sure (1) is feasible in many cases (e.g. `data_dir`), but (2) seams reasonable as long as HParams is not yet another immutable object. I was under the impression that it was, but TBF I never tested it.

Reply:
It's not immutable. It has already set and set_from_dict, and I'm not opposed to more.